### PR TITLE
Add generated types

### PR DESCRIPTION
**Pull request by nathunsmitty**
Adds generated types to the python bot.

I tried to add type hinting where it makes sense, but we won't get full type coverage at least until lidatong/dataclasses-json#23 is resolved. We still get the big win of having all dataclasses defined.

All examples and tests should work. A few dataclass field names were changed in the process.

Note that chat methods now take dataclasses and input and return dataclasses as output, as opposed to dicts as we did previously.

### CODE REVIEW

File: pykeybasebot/cli.py [hunk 1]
```diff
@@ -75,6 +76,9 @@ class KeybaseNotConnectedError(Exception):
 
     response = stdout.decode("utf-8")
     try:
-        return json.loads(response)
+        parsed_response = json.loads(response)
+        if "error" in parsed_response:
+            raise Exception(parsed_response["error"])
```
Comment (nathunsmitty) [inline pykeybasebot/cli.py]: Pass errors from the service through

File: examples/2_get_paid.py [hunk 2]
```diff
@@ -30,13 +31,17 @@ class Handler:
     async def __call__(self, bot, event):
         if event.type != EventType.WALLET:
             return
-        if event.notification.statusDescription != PaymentStatusStr.COMPLETED:
+        print(event.notification)
```
Comment (xgess) [inline examples/2_get_paid.py]: ?

File: examples/1_clap_for_everything.py [hunk 3]
```diff
@@ -10,17 +10,18 @@
...
-from pykeybasebot import Bot, ContentType
+import pykeybasebot.types.chat1 as chat1
+from pykeybasebot import Bot
 
 logging.basicConfig(level=logging.DEBUG)
 
 
 async def handler(bot, event):
-    if event.msg.content.type != ContentType.TEXT:
+    if event.msg.content.type_name != chat1.MessageTypeStrings.TEXT.value:
```
Comment (xgess) [inline examples/1_clap_for_everything.py]: `chat1.MessageTypeStrings.TEXT.value` <-- this is pretty unfortunate for something as common as this '__'

File: examples/1_clap_for_everything.py [hunk 4]
```diff
@@ -10,17 +10,18 @@
...
         return
     channel = event.msg.channel
     msg_id = event.msg.id
-    await bot.chat.react(channel.replyable_dict(), msg_id, ":clap:")
+    await bot.chat.react(channel, msg_id, ":clap:")
```
Comment (xgess) [inline examples/1_clap_for_everything.py]: this is nicer though :)

File: examples/1_pingpong.py [hunk 5]
```diff
@@ -12,18 +12,19 @@
...
+import pykeybasebot.types.chat1 as chat1
+from pykeybasebot import Bot
 
 logging.basicConfig(level=logging.DEBUG)
 
 
 class Handler:
     async def __call__(self, bot, event):
-        if event.msg.content.type != ContentType.TEXT:
+        if event.msg.content.type_name != chat1.MessageTypeStrings.TEXT.value:
```
Comment (xgess) [inline examples/1_pingpong.py]: i feel like no one is actually going to do this. they'll just write `"text"` here instead. 🙃

**Pull request merged**